pkg/skaffold/build/cluster/kaniko.go

```diff
@@ -102,3 +115,74 @@ func (b *Builder) copyKanikoBuildContext(ctx context.Context, workspace string,
 	// Generate a file to successfully terminate the init container
 	return b.kubectlcli.Run(ctx, nil, nil, "exec", podName, "-c", initContainer, "-n", b.Namespace, "--", "touch", "/tmp/complete")
 }
+
+func evaluateEnv(env []v1.EnvVar, additional ...v1.EnvVar) ([]v1.EnvVar, error) {
+	// Prepare additional envs
+	addEnv := make(map[string]string)
+	for _, addEnvVar := range additional {
+		addEnv[addEnvVar.Name] = addEnvVar.Value
+	}
+
+	// Evaluate provided env variables
+	var evaluated []v1.EnvVar
+	for _, envVar := range env {
+		val, err := util.ExpandEnvTemplate(envVar.Value, nil)
+		if err != nil {
+			return nil, fmt.Errorf("unable to get value for env variable %q: %w", envVar.Name, err)
+		}
+
+		evaluated = append(evaluated, v1.EnvVar{Name: envVar.Name, Value: val})
+
+		// Provided env variables have higher priority than additional (generated) ones
+		delete(addEnv, envVar.Name)
+	}
+
+	// Append additional (generated) env variables
+	for name, value := range addEnv {
+		if value != "" {
+			evaluated = append(evaluated, v1.EnvVar{Name: name, Value: value})
+		}
+	}
+
+	return evaluated, nil
+}
+
+func envMapFromVars(env []v1.EnvVar) map[string]string {
+	envMap := make(map[string]string)
+	for _, envVar := range env {
+		envMap[envVar.Name] = envVar.Value
+	}
+	return envMap
+}
+
+func generateEnvFromImage(imageStr string) ([]v1.EnvVar, error) {
+	repoStr, nameStr, tagStr, err := parseImageParts(imageStr)
+	if err != nil {
+		return nil, err
+	}
+	var generatedEnvs []v1.EnvVar
+	generatedEnvs = append(generatedEnvs, v1.EnvVar{Name: "IMAGE_REPO", Value: repoStr})
+	generatedEnvs = append(generatedEnvs, v1.EnvVar{Name: "IMAGE_NAME", Value: nameStr})
+	generatedEnvs = append(generatedEnvs, v1.EnvVar{Name: "IMAGE_TAG", Value: tagStr})
+	return generatedEnvs, nil
+}
+
+func parseImageParts(imageStr string) (repo, name, tag string, err error) {
```

Comment:
I am not sure i understand why we need this? We are adding extra env variables inferred from image built which are used in probably in your `Dockerfile`?
Can you elaborate more on your use case?

Thanks
Tejal

Reply:
See https://github.com/GoogleContainerTools/skaffold/issues/3229 for background information. There are quite a few use cases for having information about the image tag inside your Dockerfile. For example: you might need some kind of version tag for the application inside your docker container